src/middleware/validator.js

```diff
@@ -115,24 +95,19 @@ const validateInput = (ctx, schema) => {
   }
 }
 
-exports.validationMiddleware = directory => async (ctx, next) => {
-  if (directory) {
-    ctx.directory = directory
-    const schema = createValidationSchema(ctx)
-
-    validateInput(ctx, schema)
-    if (!ctx.input) {
-      logger.error(`Validation Failed: ${ctx.body.message}`)
-      return new Error(`Validation Failed: ${ctx.body}`)
-    }
-  } else {
-    logger.error('No input resource name provided')
-    ctx.error = 'Input resource name not given'
+exports.validateInput = validationMap => async (ctx, next) => {
+  try {
+    const schema = createValidationSchema(validationMap)
+    performValidation(ctx, schema)
```

Comment:
I think we continue to the mapping middleware even if the validation fails here. The code we had before the refactoring was checking the `ctx.input` which gets updated once validation passes. We can add back the code
`if (!ctx.input) { 
   logger.error(`Validation Failed: ${ctx.body.message}`)		
￼    return new Error(`Validation Failed: ${ctx.body}`)
}` 
Or we can throw an error when the validation fails in the performValidation function

Reply:
The `return` is in the `catch` section. So, any exception will send control back up the middleware stack.

Reply:
I've split this out into 2 separate sections. It's cleaner to check each call individually for exceptions.